share/tasks.py

```diff
@@ -86,20 +87,45 @@ def disambiguate(self, normalized_id):
     # Load all relevant ContentTypes in a single query
     ContentType.objects.get_for_models(*apps.get_models('share'), for_concrete_models=False)
 
+    updated = None
+
     try:
         with transaction.atomic():
             cg = ChangeGraph(normalized.data['@graph'], namespace=normalized.source.username)
             cg.process()
             cs = ChangeSet.objects.from_graph(cg, normalized.id)
             if cs and (normalized.source.is_robot or normalized.source.is_trusted or Source.objects.filter(user=normalized.source).exists()):
                 # TODO: verify change set is not overwriting user created object
-                cs.accept()
+                updated = cs.accept()
     except Exception as e:
         raise self.retry(
             exc=e,
             countdown=(random.random() + 1) * min(settings.CELERY_RETRY_BACKOFF_BASE ** self.request.retries, 60 * 15)
         )
 
+    # Only index creativeworks on the fly, for the moment.
+    data = [x.id for x in updated or [] if isinstance(x, AbstractCreativeWork)]
+
+    if not data:
+        return
+
+    try:
```

Comment:
This should be moved to a function (or something) in `share.search`, so it's easy to index works (or eventually anything) from anywhere.